test/arrayTest.mo

```diff
@@ -216,6 +216,11 @@ let suite = Suite.suite("Array", [
     "reverse",
     Array.reverse<Nat>([0, 1, 2, 3]),
     M.equals(T.array<Nat>(T.natTestable, [3, 2, 1, 0]))
+  ),
+  Suite.test(
+    "buffer",
+    Array.buffer<Nat>([0, 1, 2, 3]).toArray(),
```

Comment:
Are the type parameters needed?

Reply:
Hmm, I'm not sure.  I copied the code (including the type parameters) from the prior test and adapted it.  Will check.

Reply:
Looks like the type arguments are indeed required:
```
arrayTest.mo:222.5-222.31: type error [M0098], cannot implicitly instantiate function of type
  <A>[A] -> Buffer<A>
to argument of type
  [Nat]
because implicit instantiation of type parameter A is under-constrained with
  Nat  <:  A  <:  Any
where
  Nat  =/=  Any
so that explicit type instantiation is required
```
https://github.com/dfinity/motoko-base/runs/6870423809?check_suite_focus=true#step:8:373

Reply:
The type instantiation is necessary because buffer is invariant (right?) so there is no principal choice.

Reply:
I'd add a doc comment and maybe consider adding a more general 'fromIter' method too, especially if we already have  'toIter'.